### ISSUE TITLE
logits error

### ISSUE THREAD

**Issue by yushize-git** [ComDec/unirna_tf#2]
Hello! Thank you for your excellent work.When I try to get logits from the model,I found the logits so strange.I wonder whether I use the model right.
tensor([[-26.1839, -25.7726, 250.0603, -26.4847, -27.3571, 257.5663, 256.3610,
         256.5338, 259.2074, -28.4000],
        [-25.8601, -25.9026, 250.2310, -26.0428, -27.7726, 261.7134, 258.0663,
         257.8676, 257.9823, -27.6506],
        [-25.8188, -26.1926, 213.1793, -26.2648, -26.3242, 223.6499, 220.8336,
         219.6642, 219.3534, -28.9333],
        [-23.7183, -25.2815, 139.1809, -25.6983, -22.9371, 147.5416, 149.3096,
         145.2452, 145.8546, -28.2984],
        [-25.8446, -27.2529, 210.2574, -27.6332, -26.1185, 219.5518, 221.8355,
         217.7435, 218.5345, -30.2991],
        [-25.8370, -25.9290, 246.3687, -26.5832, -27.0310, 254.2923, 254.7028,
         253.3678, 256.7189, -28.4020],
        [-25.5923, -25.6817, 248.0984, -26.2822, -27.4169, 257.6693, 259.8387,
         256.7903, 257.5160, -27.6127],
        [-25.7141, -25.0530, 249.3711, -25.6346, -26.8836, 258.1454, 257.5490,
         257.3291, 259.8604, -27.2399],
        [-24.3476, -23.4631, 251.2557, -24.1338, -26.1103, 262.5418, 259.9645,
         259.8582, 260.3167, -25.4660],
        [-23.4198, -23.1718, 240.8198, -23.7626, -24.4969, 247.6712, 246.9831,
         246.7632, 248.6496, -25.9488],
        [-29.9947, -31.0068,  30.6904, -30.7071, -30.7731,  37.7138,  36.6945,
          38.6942,  37.1665, -32.4945],
        [ -5.5509,  -5.5154,  40.9718,  -5.1612,  -4.9349,  49.6181,  49.3833,
          49.3146,  50.8284,  -6.2109],
        [ -8.6768,  -9.9284, 116.1819, -10.1985,  -9.4798, 124.6332, 124.3892,
         126.0656, 124.3736, -12.8119],
        [ -8.1308, -10.0539, 136.6988, -10.4820,  -9.8236, 144.3701, 145.8625,
         144.1181, 144.3127, -12.7115],
        [ -5.4378,  -5.8618,  28.9224,  -5.3744,  -5.6022,  37.7708,  37.2874,
my code is:
import os

os.environ["HF_ENDPOINT"] = "https://hf-mirror.com"

import torch
import unirna_tf
from transformers import AutoTokenizer, AutoConfig
from unirna_tf import UniRNAForMaskedLM

pretrained_path = "./weights/unirna_L12"

tokenizer = AutoTokenizer.from_pretrained(pretrained_path)
config = AutoConfig.from_pretrained(pretrained_path)

model = UniRNAForMaskedLM.from_pretrained(pretrained_path, config=config, trust_remote_code=True)

model.eval()
device = "cuda" if torch.cuda.is_available() else "cpu"
model.to(device)

seq = "GAAUUGUGAGCGCUCACAAUUAAGGAGGAAAAAUGAAAGCUAAAGAUGUUCAGCCAACCAUUAUUAUUAAUAAAAAUGGCCUUAUCUCUUUGGAAGAUAUCUACGACAUUGCGAUGAAACAAAAAAAAGUAGAGAUAUCAACGGAGAUCACUGAACUUUUGACGUAUGGUCGUGAAAAAUUGGAGGAAAAAUUAAAUUCAGGAGAGGUUAUAUAUGGAAUCAAUACCGGAUUUGGGGGGAAUGCCAAUUUAGUUGUACCAUUUGAAAAGAUUUCAGAGCAUCAGCAAAAUCUGUUAACUUUUCUUUCUGCUGGUACUGGGGACUAUAUGUCCGAACCUUGUAUUAAAGCGUCACAAUUUACUAUGUUACUUUCUGUUUGCAAAGGUUGGUCUGCAACCAGACCAAUUGUCGCUCAGGCAAUUGUUGAUCAUAUUAAUCAUGAUAUUGUUCCUCUGGUUCCUCGCUAUGGCUCGGUGGGGGCAAGCGGCGA"
inputs = tokenizer(seq, return_tensors="pt")
inputs = {k: v.to(device) for k, v in inputs.items()}

with torch.no_grad():
    out = model(**inputs)
    logits = out.logits[0]

logits = logits[1:-1, :].float().cpu()
torch.save(logits, "unirna_logits.pt")
print(logits.shape)

**Comment by ComDec** [comment]
Hi, thanks for your feedback. UniRNAForMaskedLM is used for token prediction: given a sequence (with or without masks), it predicts the token at each position. For example, given "AU<mask>UUA", it predicts what "<mask>" could be. If you want to extract sequence embeddings, I guess "logits" that you mentioned, please use the code below:

```
import unirna_tf
from transformers import AutoTokenizer, AutoModel

source_path = "../weights/unirna_L12"
tokenizer = AutoTokenizer.from_pretrained(source_path)
model = AutoModel.from_pretrained(source_path)

seq = "GAAUUGUGAGCGCUCACAAUUAAGGAGGAAAAAUGAAAGCUAAAGAUGUUCAGCCAACCAUUAUUAUUAAUAAAAAUGGCCUUAUCUCUUUGGAAGAUAUCUACGACAUUGCGAUGAAACAAAAAAAAGUAGAGAUAUCAACGGAGAUCACUGAACUUUUGACGUAUGGUCGUGAAAAAUUGGAGGAAAAAUUAAAUUCAGGAGAGGUUAUAUAUGGAAUCAAUACCGGAUUUGGGGGGAAUGCCAAUUUAGUUGUACCAUUUGAAAAGAUUUCAGAGCAUCAGCAAAAUCUGUUAACUUUUCUUUCUGCUGGUACUGGGGACUAUAUGUCCGAACCUUGUAUUAAAGCGUCACAAUUUACUAUGUUACUUUCUGUUUGCAAAGGUUGGUCUGCAACCAGACCAAUUGUCGCUCAGGCAAUUGUUGAUCAUAUUAAUCAUGAUAUUGUUCCUCUGGUUCCUCGCUAUGGCUCGGUGGGGGCAAGCGGCGA"
token = tokenizer(seq, return_tensors="pt")
output = model(**token, output_attentions=True)
torch.save(output.last_hidden_state[1:-1].detach().cpu(), "unirna_logits.pt") # B, L, D, where L equals to sequence length
print(output.last_hidden_state.shape) # should be L+2 ([CLS] + sequence + [EOS])
```

The new version of MLM model will release ASAP. If you want to do so (token prediction), you can use it later.
